app.rb

```diff
@@ -1,5 +1,18 @@
 require_relative 'config/environment'
 
 class App < Sinatra::Base
+  get "/" do
+    erb :index
+  end
 
-end
\ No newline at end of file
+  get "/new" do
+    erb :create_puppy
+  end
+
+  post "/" do
+    @name = params[:name]
+    @age = params[:age]
+    @breed = params[:breed]
+    erb :display_puppy
+  end
```

Comment:
It would be better to use the `Puppy` class here. You could initialize it with the params, then have access it in the view.